disnake/errors.py

```diff
@@ -224,20 +224,54 @@ class ConnectionClosed(ClientException):
         The shard ID that got closed if applicable.
     """
 
+    # https://discord.com/developers/docs/topics/opcodes-and-status-codes#gateway-gateway-close-event-codes
+    GATEWAY_CLOSE_EVENT_REASONS: Dict[int, str] = {
+        4000: "Unknown error",
+        4001: "Unknown opcode",
+        4002: "Decode error",
+        4003: "Not authenticated",
+        4004: "Authentication failed",
+        4005: "Already authenticated",
+        4007: "Invalid sequence",
+        4008: "Rate limited",
+        4009: "Session timed out",
+        4010: "Invalid Shard",
+        4011: "Shard required, you are required to shard your connection in order to connect.",
+        4012: "Invalid API version",
+        4013: "Invalid intents",
+        4014: "Disallowed intents, you may have tried to specify an intent that you have not enabled or are not approved for.",
+    }
+
     def __init__(
         self,
         socket: ClientWebSocketResponse,
         *,
         shard_id: Optional[int],
         code: Optional[int] = None,
+        voice: Optional[bool] = False,
     ) -> None:
         # This exception is just the same exception except
         # reconfigured to subclass ClientException for users
         self.code: int = code or socket.close_code or -1
         # aiohttp doesn't seem to consistently provide close reason
-        self.reason: str = ""
+        if voice:
+            # https://discord.com/developers/docs/topics/opcodes-and-status-codes#voice-voice-close-event-codes
+            self.GATEWAY_CLOSE_EVENT_REASONS.update(
```

Comment:
This will change the class attribute, meaning the next call will also use these codes even if `voice=False`.
Maybe just have a separate attribute for the voice close codes? Something along the lines of:
```py
    GATEWAY_VOICE_CLOSE_EVENT_REASONS: Dict[int, str] = {
        **GATEWAY_CLOSE_EVENT_REASONS,
        4002: "Failed to decode payload",
        # ...
    }
```